compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -1514,7 +1514,8 @@ fn expr_as_for_range(
 ) -> IResult<Value> {
     expr_as(interner, arguments, return_type, location, |expr| {
         if let ExprValue::Statement(StatementKind::For(for_statement)) = expr {
-            if let ForRange::Range(from, to) = for_statement.range {
+            if let ForRange::Range(bounds) = for_statement.range {
+                let (from, to) = bounds.into_half_open();
```

Comment:
We might have to change this built-in macro method to also return a `bool` for whether the range is inclusive or not, but we can do it in a separate PR.